src/Microsoft.DotNet.Build.CloudTestTasks/CreateAzureContainer.cs

```diff
@@ -1,112 +1,185 @@
-                return false;
+                this.ReadOnlyToken = AzureHelper.CreateContainerSasToken(
+                    this.AccountName, 
+                    this.ContainerName, 
+                    this.AccountKey, 
+                    AzureHelper.SasAccessType.read, 
+                    this.ReadOnlyTokenDaysValid);
             }
-            else
+
+            if (this.WriteOnlyTokenDaysValid > 0)
             {
-                newContainer.CreateIfNotExists();
+                this.WriteOnlyToken = AzureHelper.CreateContainerSasToken(
+                    this.AccountName, 
+                    this.ContainerName, 
+                    this.AccountKey, 
+                    AzureHelper.SasAccessType.write, 
+                    this.WriteOnlyTokenDaysValid);
             }
 
-            StorageUri = newContainer.Uri.ToString();
+            this.Log.LogMessage(MessageImportance.High, "Sending request to create Container");
+            var bytestoWrite = new byte[0];
+            var bytesToWriteLength = 0;
 
-            // to keep with the msbuild convention of having a path end with
-            // a directory separator char append a '/' to the end of the URI
-            if (!StorageUri.EndsWith("/"))
-                StorageUri = string.Format("{0}/", StorageUri);
-
-            if (ReadOnlyTokenDaysValid > 0)
+            using (var postStream = await request.GetRequestStreamAsync())
             {
-                var sasRO = new SharedAccessBlobPolicy();
-                sasRO.SharedAccessExpiryTime = DateTime.UtcNow.AddDays(ReadOnlyTokenDaysValid);
-                sasRO.Permissions = SharedAccessBlobPermissions.Read;
-                ReadOnlyToken = newContainer.GetSharedAccessSignature(sasRO);
+                postStream.Write(bytestoWrite, 0, bytesToWriteLength);
             }
 
-            if (WriteOnlyTokenDaysValid > 0)
+            try
+            {
+                using (var response = await request.GetResponseAsync() as HttpWebResponse)
+                {
+                    this.Log.LogMessage(
+                        MessageImportance.High, 
+                        "Received response to create Container: Status Code: {0}", 
+                        response.StatusCode);
+                }
+            }
+            catch (WebException e)
             {
-                var sasWO = new SharedAccessBlobPolicy();
-                sasWO.SharedAccessExpiryTime = DateTime.UtcNow.AddDays(WriteOnlyTokenDaysValid);
-                sasWO.Permissions = SharedAccessBlobPermissions.Write;
-                WriteOnlyToken = newContainer.GetSharedAccessSignature(sasWO);
+                if (!e.Status.Equals("ProtocolError"))
+                {
+                    this.Log.LogError(e.Message);
+                    this.Log.LogError(e.Status.ToString());
+                    var resp = new StreamReader(e.Response.GetResponseStream()).ReadToEnd();
```

Comment:
There are 2 IDisposable things on this line. (The stream and the reader). You'll have to break it out into multiple lines to using them all correctly, unfortunately.